### PR TITLE
added unit test coverage support

### PR DESCRIPTION
**Pull request by nexus-uw**
- added unit test coverage pre test function for mocha run tests
- updated mocha to generate coverage report,
- added ensure coverage task generator
- updated README with new task creators
- pass options object to karma task to allow for karma options to be set in the user's gulpfile (like the mocha task)
- changed default mocha reporter to more professional spec reporter
- added karma coverage plugin
- added check if karma config file path is empty, so if the user wants to pass karma options directly, karma will not try to open the non-existant file

provides functionality used by an upcoming lunch and learn.

### CODE REVIEW

**Comment by nexus-uw** [comment]
Could @yuri and/or @SethDavenport take a look at this sometime this week?

File: src/rangle-gulp.js [hunk 1]
```diff
@@ -143,12 +155,61 @@ exports.mochaWatch = function (options) {
...
+  options = options || {};
+  options.thresholds = options.thresholds || {};
+
+  if(!options.coverageDirectory){
+    throw new Error('options.coverageDirectory required!')
+  }
+  var checkCoverageCommand = commandFactory.create('check-coverage');
+
+  var args=  [
+    '--statements=' + (options.thresholds.statements ? options.thresholds.statements : 100),
```
Comment (SethDavenport) [inline src/rangle-gulp.js]: 100 seems a bit high for a default threshold - I would have gone with something like 85%.

**Comment by SethDavenport** [comment]
This is good stuff.

File: src/rangle-gulp.js (hunk 1 above)
Comment (nexus-uw) [inline src/rangle-gulp.js]: go big or go home.

**Comment by yuri** [comment]
@nexus-uw @SethDavenport I think 100% makes sense as the default, if only because any other number would seem totally arbitrary. Setting the default at 100% practically means the project will have to pick a value, but at least it forces everyone to think what number they are comfortable with. So, I'll go ahead and merge.

**Pull request merged**